### PR TITLE
feat(foldable-text): Add foldable text component for FAQs etc.

### PR DESCRIPTION
**Pull request by ChefHutch**
This is a flexible, generic component for 'foldable text'

UI exampple - expanded:
Screenshot 2021-04-01 at 14 58 10

UI example - collapsed:
Screenshot 2021-04-01 at 14 58 06

Code Usage:

```js
          <FoldableText title='TITLE STRING' {...anyPropsYouWantToPassToTheWrapper}>
            {children}
          </FoldableText>
```

Code Example:
```js
 <Cardbody>
          <FoldableText title={TranslateString(999, 'Eligible trading pairs')} mt="24px>
            <Text fontSize="14px" color="textSubtle">
              {TranslateString(
                999,
                'Only trades on BNB/BUSD, CAKE/BNB, ETH/BNB and BTCB/BNB pairs will be included in volume calculations.',
              )}
            </Text>
          </FoldableText>
 </Cardbody>
```

### CODE REVIEW

File: src/components/FoldableText/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  padding-bottom: ${({ isExpanded }) => (isExpanded ? '16px' : '0px')};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.inputSecondary};
+`
+
+const FoldableText: React.FC<{ title?: string }> = ({ title, children }) => {
+  const [isExpanded, setIsExpanded] = useState(false)
+  const TranslateString = useI18n()
+
+  return (
+    <Wrapper flexDirection="column" onClick={() => setIsExpanded(!isExpanded)} mt="24px">
```
Comment (hachiojidev) [inline src/components/FoldableText/index.tsx:29]: Will this component always need 24px margin top? If this is a more generic component lets pass down the rest of the props and use it like so

```
<FoldableText mt="24px" />
```

**Review by RabbitDoge — CHANGES_REQUESTED** [review]
There is an expendableButton in the uikit. Check the variants in the storybook.

File: src/components/FoldableText/index.tsx (hunk 1 above)
Comment (ChefHutch) [reply src/components/FoldableText/index.tsx]: Nice addition man, I've made the wrapper receive `...props` & its types are extending `BoxProps`

**Comment by ChefHutch** [comment]
`Wrapper` now receives `...props` and will accept any `FlexProps` @hachiojidev 

Using ExpandableLabel from uikit @RabbitDoge 👍

File: package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     "@binance-chain/bsc-connector": "^1.0.0",
     "@crowdin/crowdin-api-client": "^1.10.4",
     "@ethersproject/abi": "^5.0.7",
-    "@pancakeswap-libs/uikit": "^0.19.0",
+    "@pancakeswap-libs/uikit": "^0.20.1",
```
Comment (ChefHutch) [inline package.json:38]: Pulling in latest uikit ver

File: src/components/FoldableText/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  height: ${({ isExpanded }) => (isExpanded ? '100%' : '0px')};
+  padding-bottom: ${({ isExpanded }) => (isExpanded ? '16px' : '0px')};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.inputSecondary};
+`
+
+const FoldableText: React.FC<FoldableTextProps> = ({ title, children, ...props }) => {
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  return (
+    <Wrapper {...props} flexDirection="column" onClick={() => setIsExpanded((prev) => !prev)}>
```
Comment (ChefHutch) [inline src/components/FoldableText/index.tsx:31]:
```suggestion
    <Wrapper {...props} flexDirection="column" onClick={() => setIsExpanded(!isExpanded)}>
```

File: src/components/FoldableText/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      <Flex justifyContent="space-between">
+        <Text fontWeight="bold" mb="16px">
+          {title}
+        </Text>
+        <StyledExpandableLabelWrapper>
+          <ExpandableLabel expanded={isExpanded} onClick={() => setIsExpanded((prev) => !prev)}>
```
Comment (ChefHutch) [inline src/components/FoldableText/index.tsx:37]:
```suggestion
          <ExpandableLabel expanded={isExpanded} onClick={() => setIsExpanded(!isExpanded)}>
```

**Review by hachiojidev — COMMENTED** [review]
Generally speaking I think components like these should be controlled meaning we should also be able to pass down the `isExpanded` prop.

**Review by hachiojidev — APPROVED** [review]

**Comment by ChefHutch** [comment]
Need to rebase to fix commit linting on some commits I did within Github interface 🤦

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**